### ISSUE TITLE
face recognition using matlab

### ISSUE THREAD

**Issue by sukesank-hub** [sukesank-hub/face-recognition-usning-matlab-#1]
clc; close all; clear;
disp('========================================');
disp('   TRAINING FACE RECOGNITION MODEL');
disp('========================================');
%% Load training data
disp('📁 Loading training images...');
imds = imageDatastore('TrainingData', ...
'IncludeSubfolders', true, ...
'LabelSource', 'foldernames');
tbl = countEachLabel(imds);
disp(tbl);
%% Split data
disp(' ');
disp('🔀 Splitting data (70% train, 30% validation)...');
[trainImds, valImds] = splitEachLabel(imds, 0.7, 'randomized');
%% Load pretrained AlexNet
disp(' ');
disp('📦 Loading pretrained AlexNet...');
net = alexnet;
inputSize = net.Layers(1).InputSize;
% Replace final layers
layersTransfer = net.Layers(1:end-3);
numClasses = numel(categories(trainImds.Labels));
layers = [
    layersTransfer
    fullyConnectedLayer(numClasses, 'WeightLearnRateFactor', 20, 'BiasLearnRateFactor', 20)
    softmaxLayer
    classificationLayer];
%% Add data augmentation
disp(' ');
disp('🖼️  Setting up data augmentation...');
imageAugmenter = imageDataAugmenter( ...
'RandRotation', [-20, 20], ...
'RandXTranslation', [-10 10], ...
'RandYTranslation', [-10 10], ...
'RandXScale', [0.9 1.1], ...
'RandYScale', [0.9 1.1]);
augmentedTrainImds = augmentedImageDatastore(inputSize(1:2), trainImds, ...
'DataAugmentation', imageAugmenter, ...
'ColorPreprocessing', 'gray2rgb');
augmentedValImds = augmentedImageDatastore(inputSize(1:2), valImds, ...
'ColorPreprocessing', 'gray2rgb');
%% Training options
options = trainingOptions('sgdm', ...
'MiniBatchSize', 8, ...
'MaxEpochs', 15, ...
'InitialLearnRate', 1e-4, ...
'LearnRateSchedule', 'piecewise', ...
'LearnRateDropFactor', 0.5, ...
'LearnRateDropPeriod', 5, ...
'L2Regularization', 0.0001, ...
'Shuffle', 'every-epoch', ...
'ValidationData', augmentedValImds, ...
'ValidationFrequency', 5, ...
'ValidationPatience', 5, ...
'Verbose', true, ...
'Plots', 'training-progress');
%% Train the network
disp(' ');
disp('🎓 Training network...');
disp('   This may take 5-10 minutes...');
myNet = trainNetwork(augmentedTrainImds, layers, options);
%% Test accuracy
disp(' ');
disp('✅ Training complete! Testing accuracy...');
[YPred, scores] = classify(myNet, augmentedValImds);
YValidation = valImds.Labels;
accuracy = sum(YPred == YValidation) / numel(YValidation) * 100;
fprintf('   Validation Accuracy: %.2f%%\n', accuracy);
% Show confusion matrix
figure;
confusionchart(YValidation, YPred);
title('Confusion Matrix');
%% Save model
disp(' ');
disp('💾 Saving trained model...');
save('myNet.mat', 'myNet', '-v7.3');
disp('✅ Model saved as myNet.mat');
disp('========================================');
disp('   TRAINING COMPLETE!');
disp('   Now run face_det.m for detection');
disp('========================================');
